smac/intensification/intensification.py

```diff
@@ -137,30 +192,303 @@ def __init__(self, tae_runner: ExecuteTARun,
+                     ) -> typing.Tuple[IntensifierBehest, RunInfo]:
+        """
+        This procedure is in charge of generating a RunInfo object to comply
+        with lines 7 (in case stage is stage==RUN_INCUMBENT) or line 12
+        (In case of stage==RUN_CHALLENGER)
+
+        A RunInfo object encapsulates the necessary information for a worker
+        to execute the job, nevertheless, a challenger is not always available.
+        This could happen because no more configurations are available or the new
+        configuration to try was already executed.
+
+        To circumvent this, a behest is also returned:
+
+        - (behest=RUN) Run the RunInfo object (Normal Execution
+        - (behest=TERMINATE) No more challengers available (Line 19 should occur)
+        - (behest=SKIP) Skip this iteration. No challenger is available, in particular
+            because challenger is the same as incumbent
+
+
+        Parameters
+        ----------
+        challengers : typing.List[Configuration]
+            promising configurations
+        incumbent: Configuration
+            incumbent configuration
+        chooser : smac.optimizer.epm_configuration_chooser.EPMChooser
+            optimizer that generates next configurations to use for racing
+        run_history : typing.Optional[smac.runhistory.runhistory.RunHistory]
+            stores all runs we ran so far
+        repeat_configs : bool
+            if False, an evaluated configuration will not be generated again
+
+        Returns
+        -------
+        behest: IntensifierBehest
+            What should the smbo object do with the runinfo.
+        run_info: RunInfo
+            An object that encapsulates necessary information for a config run
+        """
+
+        # In case a crash happens, and FirstRunCrashedException prevents a
+        # failure, revert back to running the incumbent
+        # Challenger case is by construction ok, as there is no special
+        # stage for its processing
+        if self.stage == IntensifierStage.PROCESS_FIRST_CONFIG_RUN:
+            self.stage = IntensifierStage.RUN_FIRST_CONFIG
+        elif self.stage == IntensifierStage.PROCESS_INCUMBENT_RUN:
+            self.stage = IntensifierStage.RUN_INCUMBENT
+
+        try:
+
+            # Line 2:
+            # Understand who is the challenger.
+            # Not necessarily a challenger is always a new configuration,
+            # as this method might be called multiple times to
+            # gain more confidence on incumbent, or even run the same
+            # challenger on multiple N (line 17)
+            challenger = self.query_challenger(
+                challengers=challengers,
+                chooser=chooser,
+                run_history=run_history,
+                repeat_configs=repeat_configs,
+            )
+
+            # if first ever run, then assume current challenger to be the incumbent
+            # In this case, we need a branch new challenger
```

Comment:
What's a 'branch new challenger'?

Reply:
Updated the comment, it was supposed to be brand, but moved away from this.